src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeTools.java

```diff
@@ -31,13 +35,25 @@
     public void Replace(String replacement, TaskListener listener) throws IOException, InterruptedException {
         if (replacement != null && !replacement.isEmpty())
         {
-            String content = file.readToString();  // needs to use read() instead!
+            BOMInputStream inputStream = new BOMInputStream(file.read());
+            ByteOrderMark bom = inputStream.getBOM();
+
+            Charset fileEncoding = Charset.defaultCharset();
+            if (bom != null) {
+                fileEncoding = Charset.forName(bom.getCharsetName());
+            }
+
+            String content = IOUtils.toString(inputStream, fileEncoding);
+            inputStream.close();
```

Comment:
There should probably be a try/finally around this so that the stream gets closed.